lib/utils.js

```diff
@@ -15,71 +22,83 @@ export function normalizePredicate(predicate) {
   return isInverse(predicate) ? predicate.slice(1) : predicate;
 }
 
-export function serializeTriple(triple) {
+export function serializeTriple(triple, {close = true} = {}) {
   const predicate = sparqlEscapePredicate(triple.predicate.value);
-  return `${serializeTriplePart(triple.subject)} ${predicate} ${serializeTriplePart(triple.object)}.`;
+  return `${serializeTriplePart(triple.subject)} ${predicate} ${serializeTriplePart(triple.object)}${close ? '.' : ''}`;
 }
 
-export function serializeTriplePart(triplePart){
-  if(triplePart.type == 'uri' || triplePart.termType == "NamedNode"){
+export function serializeTriplePart(triplePart) {
+  if (triplePart.type == 'uri' || triplePart.termType == 'NamedNode') {
     return sparqlEscapeUri(triplePart.value);
-  }
-  else if (triplePart.type === 'literal' || triplePart.type === 'typed-literal') {
-    if(triplePart.datatype) {
-        return `${sparqlEscapeString(triplePart.value)}^^${sparqlEscapeUri(triplePart.datatype)}`;
-    }
-    else if(triplePart.lang) {
+  } else if (triplePart.type === 'literal' || triplePart.type === 'typed-literal') {
+    if (triplePart.datatype) {
+      return `${sparqlEscapeString(triplePart.value)}^^${sparqlEscapeUri(triplePart.datatype)}`;
+    } else if (triplePart.lang) {
       return `${sparqlEscapeString(triplePart.value)}@${triplePart.lang}`;
-    }
-    else {
+    } else {
       return sparqlEscapeString(triplePart.value);
     }
-  }
-  else {
+  } else {
     console.log(`Don't know how to escape type ${triplePart.type}. Will escape as a string.`);
     return sparqlEscapeString(triplePart.value);
   }
 }
 
 /**
  * Returns whether 2 arrays of path segments (as string) are equal
-*/
+ */
 export function isSamePath(a, b) {
   return a.join('/') == b.join('/');
 }
 
+export function compareTriple(t1, t2) {
```

Comment:
There might some cases where this fails:
- e.g.  what if object.value is the same, but the data types are different
- and a very subtle one: logical equivalents of lexical different representations, see e.g. https://github.com/lblod/delta-producer-publication-graph-maintainer/blob/master/publication-graph-maintenance/main.js#L115 and https://github.com/lblod/delta-producer-publication-graph-maintainer/blob/master/jobs-processing/pipeline-healing.js#L253